<?xml version="1.0"?>

<launch>
    <!-- run pointcloud_to_laserscan node -->
    <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan">

        <remap from="cloud_in" to="/velodyne_points"/>
        <remap from="scan" to="/velodyne_2dscan"/>
        <rosparam>
            target_frame: base_link
            transform_tolerance: 0.01
            min_height: 0.75
            max_height: 10

            angle_min: -3.14 # -M_PI
            angle_max: 3.14 # M_PI
            angle_increment: 0.00349066 #  0.2 deg
            scan_time: 0.1 # 0.05
            range_min: 0.4
            range_max: 130.0
            use_inf: true

            # Concurrency level, affects number of pointclouds queued for processing and number of threads used
            # 0 : Detect number of cores
            # 1 : Single threaded
            # 2->inf : Parallelism level
            concurrency_level: 1
        </rosparam>

    </node>

</launch>
